cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/SystemArguments.java

```diff
@@ -357,12 +357,23 @@ public static Resources getResources(Map<String, String> args, @Nullable Resourc
    * @return the profile id for the program run
    */
   public static ProfileId getProfileIdForProgram(ProgramId programId, Map<String, String> args) {
-    if (programId.getType() == ProgramType.WORKFLOW) {
+    if (isProfileAllowed(programId.getType())) {
       return getProfileIdFromArgs(programId.getNamespaceId(), args).orElse(ProfileId.NATIVE);
     }
     return ProfileId.NATIVE;
   }
 
+  /**
+   * Determine if compute profile is allowed to be applied on the given program type.
```

Comment:
nit: Add a new line after this for keeping it consistent with other comments?

Reply:
Done